addons/sourcemod/scripting/vsh/bosses/boss_announcer.sp

```diff
@@ -379,13 +447,35 @@ public Action Timer_AnnouncerChangeTeam(Handle hTimer, int iClient)
 		return;
 	}
 	
-	PrintCenterText(iClient, "YOU'RE NOW IN BOSS TEAM");
+	//Need to detach buildings from engineers before switching teams so they don't explode
+	if (TF2_GetPlayerClass(iClient) == TFClass_Engineer)
```

Comment:
Engineer class check shouldn't be needed when we want to imagine VSH + Randomizer combined